### PR TITLE
[GUI] Lock wallet after TX if unlocked for staking only

### PR DESCRIPTION
**Pull request by akshaynexus**
This PR adds a check to the send function of the Qt send UI,to lock wallet after sending out a tx if wallet was unlocked for staking only.Also modifies the setwalletunlocked function in walletmodel to use the anonymizeonly boolean from its function while setting the walletmodel's fWalletUnlockAnonymizeOnly bool.

### CODE REVIEW

**Comment by furszy** [comment]
Please clean the commits history, pure merge commits.

File: src/qt/pivx/send.cpp [hunk 1]
```diff
@@ -370,6 +370,11 @@ bool SendWidget::send(QList<SendCoinsRecipient> recipients){
         if (sendStatus.status == WalletModel::OK) {
             clearAll();
             inform(tr("Transaction sent"));
+            //Check if wallet is unlocked for staking
+            if (walletModel->getEncryptionStatus() == WalletModel::UnlockedForAnonymizationOnly) {
+                //Lock wallet after sending out a Transaction
+                walletModel->setWalletLocked(true,SecureString(),true);
```
Comment (furszy) [inline src/qt/pivx/send.cpp]: This is not going to work, if the software gets to that line of code then the wallet is fully unlocked and never in UnlockedForAnonymizationOnly state (the whole point of UnlockedForAnonymizationOnly is only let the wallet stake and block every other action).

To implement it correctly follow the contextual unlock/lock flow.
(which i'm pretty sure that we don't have it finished and needs to be back ported).
 Check inside the walletModel file, UnlockContext object. Methods requestUnlock & UnlockContext.

Essentially, it's an abstraction to acquire the wallet unlock for an specific action and once it's completed re lock it.

Comment (akshaynexus) [reply src/qt/pivx/send.cpp]: Alright,will redo the implementation.

**Comment by akshaynexus** [comment]
closing this temporarily,until i backport the unlockcontext code.

**Pull request closed without merging**